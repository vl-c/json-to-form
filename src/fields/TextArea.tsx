import * as React from 'react';

export interface TextAreaProps {
  rows?: number;
  id: string;
  label?: string;
  value?: string;
  required?: boolean;
  disabled?: boolean;
  readonly?: boolean;
}

const TextArea = (props: TextAreaProps): JSX.Element => {
  const { id, rows, label, value, readonly, disabled } = props;

  return (
    <div className="input-block">
      {label && (
        <label className="input-label" htmlFor={id}>
          {label}
        </label>
      )}
      <textarea
        className="input-field"
        id={id}
        name={id}
        rows={rows}
        readOnly={readonly}
        disabled={disabled}
        defaultValue={value}
      />
    </div>
  );
};

export default TextArea;
