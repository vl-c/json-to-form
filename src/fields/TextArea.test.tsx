import * as React from 'react';

import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import TextArea from './TextArea';

function renderTextArea() {
  return shallow(
    <TextArea
      id={'test'}
      label={'TextArea'}
      value={
        'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad culpa eveniet id maxime nobis, tenetur.'
      }
    />
  );
}

it('TextArea shallow render', () => {
  const res = renderTextArea();
  expect(shallowToJson(res)).toMatchSnapshot();
});
