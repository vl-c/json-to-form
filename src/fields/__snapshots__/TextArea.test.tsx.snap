// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea shallow render 1`] = `
<div
  className="input-block"
>
  <label
    className="input-label"
    htmlFor="test"
  >
    TextArea
  </label>
  <textarea
    className="input-field"
    defaultValue="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad culpa eveniet id maxime nobis, tenetur."
    id="test"
    name="test"
  />
</div>
`;
